packages/flutter_test/lib/src/finders.dart

```diff
@@ -583,3 +606,37 @@ class _DescendantFinder extends Finder {
     return candidates;
   }
 }
+
+class _AncestorFinder extends Finder {
+  _AncestorFinder(this.descendant, this.ancestor, { this.matchRoot: false }) : super(skipOffstage: false);
+
+  final Finder ancestor;
+  final Finder descendant;
+  final bool matchRoot;
```

Comment:
I guess there's already prior art, but 'root' is a weird word. Maybe matchSelf is clearer

Reply:
I suppose I was trying to be consistent with find.descendant since they're analogs of each other.